cmd/zpool/zpool.d/media

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+#
+# Print out the type of device
+#
+
+helpstr="
+media:		Show whether a vdev is a file, hdd, or ssd."
+
+script=$(basename "$0")
+if [ "$1" = "-h" ] ; then
+	echo "$helpstr" | grep "$script:" | tr -s '\t' | cut -f 2-
+	exit
+fi
+
+file_type=$(file -b "$VDEV_UPATH" 2>/dev/null)
+
+if [ "$file_type" = "block special" ]; then
+	device=$(basename "$VDEV_UPATH")
+	val=$(cat "/sys/block/$device/queue/rotational" 2>/dev/null)
+	if [ "$val" = "0" ]; then
+		echo "media=ssd"
```

Comment:
Just assign `media=<type>` in the if blocks and then at the end `echo "media=$media".`  That removes the need for the exits.